### ISSUE TITLE
querySelector  not work for attr whose value contains colon

### ISSUE THREAD

**Issue by ilyydy** [capricorn86/happy-dom#632]
**Describe the bug**
querySelector  not work for attr whose value contains colon. But in browser, it will be ok.

**To Reproduce**
```js
import { Window } from 'happy-dom';

const window = new Window();
const document = window.document;

document.body.innerHTML = `<meta ab="a:b"/>`;
const element = document.querySelector('[ab="a\\:b"]');
console.log(element !== null); // output false
const element1 = document.querySelector('[ab="a:b"]');
console.log(element1 !== null); // output false

```

**Expected behavior**
should find meta element

**Screenshots**

**Device:**
 - OS: win10
 - Browser: chrome
 - Version: 106

**Additional context**

**Comment by Pumpuli** [comment]
This is actually quite a big problem when `happy-dom` is used for testing React applications. That's because React's `useId()` hook returns strings which always seem to contain colon characters. And even if `useId()` is not used directly, for example Material UI's form field components use it internally. The typical use case for `useId()` is form controls and their labels:

```js
function PasswordInput() {
  const suffix = useId();

  return (
    <>
      <label>
        Password:
        <input type="password" aria-describedby={`password-${suffix}`} />
      </label>
      <p id={`password-${suffix}`}>Should contain at least 18 characters.</p>
    </>
  );
}
```

The result of that is something like `aria-describedby="password-:r5:"` and `id="password-:r5:"`. `@testing-library/dom` then [tries to do](https://github.com/testing-library/dom-testing-library/blob/fb069c93983bc0300a6e1c91bdec5bf9443b5286/src/label-helpers.ts#L66-L68) `.querySelector('[id="password-:r5:"]')`, which yields `null` on `happy-dom`. The problem can be traced back to the regex that checks whether a selector has a pseudo class:

https://github.com/capricorn86/happy-dom/blob/13bcfe77ae9a202a93ce8fddcf4a7c1b43ecde16/packages/happy-dom/src/query-selector/SelectorItem.ts#L8-L9

However, I'm not sure what's the best way to go about fixing this. Should the regex be made more complicated to make sure that an attribute selector is excluded?

**Comment by capricorn86** [comment]
Thank you for reporting @ilyydy and @Pumpuli ! :slightly_smiling_face: 

I believe that the query selector for `[ab="a:b"]` was fixed in [v9.10.0](https://github.com/capricorn86/happy-dom/releases/tag/v9.10.0).

I have now added support for using an escape character, so that the `[ab="a\\:b"]` query selector will work.

You can read more about the release here:
https://github.com/capricorn86/happy-dom/releases/tag/v10.5.2
